pkg/cosmos/client/gas_price_estimator.go

```diff
@@ -17,16 +19,77 @@ var _ GasPricesEstimator = (*FixedGasPriceEstimator)(nil)
 
 type FixedGasPriceEstimator struct {
 	gasPrices map[string]sdk.DecCoin
+	lggr      logger.SugaredLogger
 }
 
-func NewFixedGasPriceEstimator(prices map[string]sdk.DecCoin) *FixedGasPriceEstimator {
-	return &FixedGasPriceEstimator{gasPrices: prices}
+func NewFixedGasPriceEstimator(prices map[string]sdk.DecCoin, lggr logger.SugaredLogger) *FixedGasPriceEstimator {
+	return &FixedGasPriceEstimator{gasPrices: prices, lggr: lggr}
 }
 
 func (gpe *FixedGasPriceEstimator) GasPrices() (map[string]sdk.DecCoin, error) {
 	return gpe.gasPrices, nil
 }
 
+func (gpe *FixedGasPriceEstimator) GasPrice(coin string) sdk.DecCoin {
+	return gpe.gasPrices[coin]
+}
+
+func (gpe *FixedGasPriceEstimator) SetGasPrice(coin string, price sdk.DecCoin) {
+	gpe.gasPrices[coin] = price
+}
+
+// CalculateGasPrice calculates the minimum of the maximum input gas prices.
+// Parameters:
+// - maxGasPrice: max gas price set by the user or operator
+// - defaultGasPrice: default gas price
+// - maxGasPriceConfigured: max gas price originally configured for the node
+func (gpe *FixedGasPriceEstimator) CalculateGasPrice(
```

Comment:
```
// - maxGasPrice: max gas price set by the user or operator
// - defaultGasPrice: default gas price
// - maxGasPriceConfigured: max gas price originally configured for the node
func (gpe *FixedGasPriceEstimator) CalculateGasPrice(
```

Currently, this is not being called anywhere. initially i was thinking of not having this in. However i thought that this might be useful in the future if we would like to calculate the gas price based on parameters instead of hard coding the fixed price value of `FallbackGasPrice() sdk.Dec` in for fixed price estimator. 

So instead of using `FallbackGasPrice() sdk.Dec`, we can pass in maxGasPrice for each transaction, and maxGasPriceConfigured by the node along with `FallbackGasPrice() sdk.Dec`, to get a gas price?

This could be relevant in the case of cosmos gas when `min-gas-prices` is a parameter local to each full-node? 

Do you think this part would be useful, or would it be clearer to remove it @cfal ?

Reply:
that make sense, we can always implement them if needed afterwards. Thanks for the suggestion!